### DISCUSSION TITLE
Song Select filtering is not respected in certain scenarios

### DISCUSSION THREAD

**Discussion by LeNitrous** [ppy/osu#19438] [Q&A] [answered]
### Type

Game behaviour

### Bug description

In song select, using either the "Next" or "Previous" track keys or pressing the respective buttons through the music controller ignores song select filtering (collection, search terms, ruleset) making beatmaps show up in the carousel.

Related to https://github.com/ppy/osu/issues/11325 (using Go to beatmap inserts the said beatmap to the carousel).

### Screenshots or videos

https://user-images.githubusercontent.com/20495991/181449684-620cf10b-1db7-438a-a496-5751f1f5a38b.mp4

### Version

2022.733.0-lazer

### Logs

[network.log](https://github.com/ppy/osu/files/9206736/network.log)
[performance.log](https://github.com/ppy/osu/files/9206737/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/9206738/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/9206739/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/9206740/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/9206741/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/9206742/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9206743/updater.log)
[database.log](https://github.com/ppy/osu/files/9206744/database.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/9206745/legacy-ipc.log)

**Comment by peppy** [comment, marked as answer]
This is intentional, to allow displaying a current selection that the user has requested when it is otherwise filtered away.

Is this in some way annoying to you?

**Reply by LeNitrous** [reply]
I was expecting that it will cycle through the currently displayed beatmaps. Also the current ruleset can get changed in the process which is annoying.

**Reply by peppy** [reply]
That's not the intention of the music controller. It has its own filtering mechanism and is not coupled to song select.
